consensus/src/processes/pruning_proof/mod.rs

```diff
@@ -656,32 +665,39 @@ impl PruningProofManager {
             .collect_vec()
     }
 
-    fn block_at_depth(&self, ghostdag_store: &impl GhostdagStoreReader, high: Hash, depth: u64) -> Hash {
-        let high_gd = ghostdag_store.get_compact_data(high).unwrap();
+    fn block_at_depth(&self, ghostdag_store: &impl GhostdagStoreReader, high: Hash, depth: u64) -> Result<Hash, String> {
```

Comment:
It's best practice to return an error that implements the `Error` trait. You can also keep the previous function signature and use `unwrap_or_else` instead of `map_err` so you can format the panic message. (same comment for the other functions)

Reply:
I want the panic at the caller for additional info. Added a small Error wrapper enum